src/v2/System/useFeatureFlag.tsx

```diff
@@ -55,36 +51,40 @@ export function useTrackVariantView({
   experimentName,
   variantName,
   payload,
-  contextOwnerType,
-  contextOwnerId,
-  contextOwnerSlug,
-  shouldTrackExperiment = true,
 }: VariantTrackingProperties) {
-  const { trackEvent } = useTracking()
-
-  useEffect(() => {
-    if (typeof window !== "undefined") {
-      const trackFeatureView = shouldTrack(experimentName, variantName)
-
-      if (trackFeatureView && shouldTrackExperiment) {
-        trackEvent({
-          action: ActionType.experimentViewed,
-          service: "unleash",
-          experiment_name: experimentName,
-          variant_name: variantName,
-          payload,
-          context_owner_type: contextOwnerType,
-          context_owner_id: contextOwnerId,
-          context_owner_slug: contextOwnerSlug,
-        })
-      }
+  const router = useRouter()
+
+  const trackVariantView = () => {
+    // HACK: Temporary hack while we refactor useAnalyticsContext
+    // to update upon page navigation.
+    const path = router.match.location.pathname
+    const pageParts = path.split("/")
+    const pageSlug = pageParts[2]
+    const pageType = formatOwnerTypes(path)
+
+    const trackFeatureView = shouldTrack(experimentName, variantName)
+
+    if (trackFeatureView) {
+      // HACK: We are using window.analytics.track over trackEvent from useTracking because
+      // the trackEvent wasn't behaving as expected, it was never firing the event and
+      // moving to using the solution below fixed the issue.
+      window?.analytics?.track(ActionType.experimentViewed, {
```

Comment:
See discussion [here](https://github.com/artsy/force/pull/9679#discussion_r834230709) resulting in this hack